File: go-client-sdk/pkg/models/shared/codatpublicapimodelsdatatrackingcategorytree.go
package shared

import (
	"time"
)

type CodatPublicAPIModelsDataTrackingCategoryTree struct {
	HasChildren        *bool                                                 `json:"hasChildren,omitempty"`
	ID                 *string                                               `json:"id,omitempty"`
	ModifiedDate       *time.Time                                            `json:"modifiedDate,omitempty"`
	Name               *string                                               `json:"name,omitempty"`
	ParentID           *string                                               `json:"parentId,omitempty"`
	SourceModifiedDate *time.Time                                            `json:"sourceModifiedDate,omitempty"`
	Status             *CodatDataContractsDatasetsTrackingCategoryStatusEnum `json:"status,omitempty"`
	SubCategories      []CodatPublicAPIModelsDataTrackingCategoryTree        `json:"subCategories,omitempty"`
}

File: python-client-sdk/src/sdk/models/shared/codat_public_api_models_data_trackingcategorytree.py
import dataclasses
import dateutil.parser
from ..shared import codat_datacontracts_datasets_trackingcategorystatus_enum as shared_codat_datacontracts_datasets_trackingcategorystatus_enum
from dataclasses_json import dataclass_json
from datetime import datetime
from marshmallow import fields
from sdk import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class CodatPublicAPIModelsDataTrackingCategoryTree:
    has_children: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('hasChildren') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    modified_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('modifiedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    parent_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('parentId') }})
    source_modified_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sourceModifiedDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    status: Optional[shared_codat_datacontracts_datasets_trackingcategorystatus_enum.CodatDataContractsDatasetsTrackingCategoryStatusEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('status') }})
    sub_categories: Optional[list[CodatPublicAPIModelsDataTrackingCategoryTree]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('subCategories') }})
    